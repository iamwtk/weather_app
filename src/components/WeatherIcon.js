import React, { Component } from 'react';

/**
 * Loading SVG icons as Component via svg-react-loader in webpack
 * @type {Component}
 */
import IconDay01 from '-!svg-react-loader?name=MyIcon!../icons/day01.svg'
import IconNight01 from '-!svg-react-loader?name=MyIcon!../icons/night01.svg'
import IconDay02 from '-!svg-react-loader?name=MyIcon!../icons/day02.svg'
import IconNight02 from '-!svg-react-loader?name=MyIcon!../icons/night02.svg'
import IconDay03 from '-!svg-react-loader?name=MyIcon!../icons/day03.svg'
import IconNight03 from '-!svg-react-loader?name=MyIcon!../icons/night03.svg'
import IconDay04 from '-!svg-react-loader?name=MyIcon!../icons/day04.svg'
import IconNight04 from '-!svg-react-loader?name=MyIcon!../icons/night04.svg'
import IconDay09 from '-!svg-react-loader?name=MyIcon!../icons/day09.svg'
import IconNight09 from '-!svg-react-loader?name=MyIcon!../icons/night09.svg'
import IconDay10 from '-!svg-react-loader?name=MyIcon!../icons/day10.svg'
import IconNight10 from '-!svg-react-loader?name=MyIcon!../icons/night10.svg'
import IconDay11 from '-!svg-react-loader?name=MyIcon!../icons/day11.svg'
import IconNight11 from '-!svg-react-loader?name=MyIcon!../icons/night11.svg'
import IconDay13 from '-!svg-react-loader?name=MyIcon!../icons/day13.svg'
import IconNight13 from '-!svg-react-loader?name=MyIcon!../icons/night13.svg'
import IconDay50 from '-!svg-react-loader?name=MyIcon!../icons/day50.svg'
import IconNight50 from '-!svg-react-loader?name=MyIcon!../icons/night50.svg'

class WeatherIcon extends Component {

  constructor(props) {
    super(props)
    /**
     * Assigns SVG icon modules to icon codes received from OpenWeatherMap API
     * @type {Object}
     */
    this.icons = {
      '01d': IconDay01,
      '01n': IconNight01,
      '02d': IconDay02,
      '02n': IconNight02,
      '03d': IconDay03,
      '03n': IconNight03,
      '04d': IconDay04,
      '04n': IconNight04,
      '09d': IconDay09,
      '09n': IconNight09,
      '10d': IconDay10,
      '10n': IconNight10,
      '11d': IconDay11,
      '11n': IconNight11,
      '13d': IconDay13,
      '13n': IconNight13,
      '50d': IconDay50,
      '50n': IconNight50
    }
  }

  render() {
    /**
     * Gets Component from this.icons Object by key passed as a prop to WeatherIcon Component
     * then renders Component
     * if icon was not passed defaults to IconDay1
     * @type {Component}
     */
    let IconName;
    console.log(this.props);
    this.props.icon ?
      IconName = this.icons[this.props.icon] :
      IconName = IconDay01



    return <IconName className={this.props.addClass ? this.props.addClass : 'null'}/>

  }

}

export default WeatherIcon;
