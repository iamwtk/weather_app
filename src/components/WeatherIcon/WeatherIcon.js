import React, { Component } from 'react';

/**
 * Loading SVG icons as Component via svg-react-loader in webpack
 * @type ReactComponent
 */
import IconDay01 from '-!svg-react-loader?name=MyIcon!../../icons/day01.svg'
import IconNight01 from '-!svg-react-loader?name=MyIcon!../../icons/night01.svg'
import IconDay02 from '-!svg-react-loader?name=MyIcon!../../icons/day02.svg'
import IconNight02 from '-!svg-react-loader?name=MyIcon!../../icons/night02.svg'
import IconDay03 from '-!svg-react-loader?name=MyIcon!../../icons/day03.svg'
import IconNight03 from '-!svg-react-loader?name=MyIcon!../../icons/night03.svg'
import IconDay04 from '-!svg-react-loader?name=MyIcon!../../icons/day04.svg'
import IconNight04 from '-!svg-react-loader?name=MyIcon!../../icons/night04.svg'
import IconDay09 from '-!svg-react-loader?name=MyIcon!../../icons/day09.svg'
import IconNight09 from '-!svg-react-loader?name=MyIcon!../../icons/night09.svg'
import IconDay10 from '-!svg-react-loader?name=MyIcon!../../icons/day10.svg'
import IconNight10 from '-!svg-react-loader?name=MyIcon!../../icons/night10.svg'
import IconDay11 from '-!svg-react-loader?name=MyIcon!../../icons/day11.svg'
import IconNight11 from '-!svg-react-loader?name=MyIcon!../../icons/night11.svg'
import IconDay13 from '-!svg-react-loader?name=MyIcon!../../icons/day13.svg'
import IconNight13 from '-!svg-react-loader?name=MyIcon!../../icons/night13.svg'
import IconDay50 from '-!svg-react-loader?name=MyIcon!../../icons/day50.svg'
import IconNight50 from '-!svg-react-loader?name=MyIcon!../../icons/night50.svg'


/*
 * WeatherIcon Component
 * renders SVG icon based on icon code passed via props
 */
class WeatherIcon extends Component {
  /**
   * Sets icon component based on icon id string passed as props
   * @param  {String} icon string of icon id passed as props
   * @return {ReactComponent} ReactComponent matching icon id
   */
  getIconComponent(icon) {
    const icons = {
      '01d': IconDay01,
      '01n': IconNight01,
      '02d': IconDay02,
      '02n': IconNight02,
      '03d': IconDay03,
      '03n': IconNight03,
      '04d': IconDay04,
      '04n': IconNight04,
      '09d': IconDay09,
      '09n': IconNight09,
      '10d': IconDay10,
      '10n': IconNight10,
      '11d': IconDay11,
      '11n': IconNight11,
      '13d': IconDay13,
      '13n': IconNight13,
      '50d': IconDay50,
      '50n': IconNight50
    }
    return icons[icon]
  }
  /**
   * Component Render Method
   * @return ReactElement markup
   */
  render() {
    /**
     * Checks if icon id was passed as props
     * if yes calls getIconComponent with icon id as parameter
     * if not defaults to IconDay01
     * @type {ReactComponent}
     */
    let IconName
    this.props.icon ?
      IconName = this.getIconComponent(this.props.icon) :
      IconName = IconDay01

    return <IconName className={this.props.addClass ? this.props.addClass : 'null'}/>

  }

}

export default WeatherIcon
