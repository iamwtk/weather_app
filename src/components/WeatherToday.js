import React from 'react';

import WeatherIcon from '../components/WeatherIcon';

import style from './weatherToday.scss';

/**
 * Renders Widget for todays weather
 * @param props passed from ../containers/WeatherScreen
 */
const WeatherToday = (props) => (
  <div className={style.weather_today}>
    <div className={style.col_left}>
      <span className={style.day_name}>{props.data.day}</span>
      <span className={style.date}>{props.data.date}</span>
    </div>
    <div className={style.col_right}>
      <span className={style.temp}>{ Math.round(props.data.temp)}&#176;</span>
      <span className={style.place}>{props.data.place}</span>
      <WeatherIcon addClass={style.weather_icon} icon={props.data.icon} />
    </div>
  </div>
);

export default WeatherToday;
