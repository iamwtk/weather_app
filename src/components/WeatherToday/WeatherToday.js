import React from 'react'

import WeatherIcon from '../../components/WeatherIcon/WeatherIcon'

import style from './weatherToday.scss'

/**
 * Renders Widget for todays weather
 * @param props passed from ../containers/WeatherScreen
 */
const WeatherToday = (props) => {
  //Destructuring of props
  const { day, date, temp, place, icon } = props.data
  return (
    <div className={style.weather_today}>
      <div className={style.col_left}>
        <span className={style.day_name}>{day}</span>
        <span className={style.date}>{date}</span>
      </div>
      <div className={style.col_right}>
        <span className={style.temp}>{ Math.round(temp)}&#176;</span>
        <span className={style.place}>{place}</span>
        <WeatherIcon addClass={style.weather_icon} icon={icon} />
      </div>
    </div>
  )
}
export default WeatherToday
