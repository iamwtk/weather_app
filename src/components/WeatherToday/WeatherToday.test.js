import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import WeatherToday from './WeatherToday'

import WeatherIcon from '../../components/WeatherIcon/WeatherIcon'

import style from './weatherToday.scss'


describe('<WeatherToday/>', () => {
  const data = {
    day: 'Monday',
    date: 'Oct. 25',
    temp: 32.25,
    place: 'Liberec, CZ',
    icon: '01n'
  }
  let component
  beforeEach(() => {
    component = shallow(<WeatherToday data={data}/>)
  })
  describe('Snapshot', () => {
    it('Should match snapshot', () => {
      const tree = toJson(component)
      expect(tree).toMatchSnapshot()
    })
  })
})
