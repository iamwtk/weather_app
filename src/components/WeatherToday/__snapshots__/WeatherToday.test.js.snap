// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherToday/> Snapshot Should match snapshot 1`] = `
<div
  className="weather_today"
>
  <div
    className="col_left"
  >
    <span
      className="day_name"
    >
      Monday
    </span>
    <span
      className="date"
    >
      Oct. 25
    </span>
  </div>
  <div
    className="col_right"
  >
    <span
      className="temp"
    >
      32
      Â°
    </span>
    <span
      className="place"
    >
      Liberec, CZ
    </span>
    <WeatherIcon
      addClass="weather_icon"
      icon="01n"
    />
  </div>
</div>
`;
